kit/controls/src/UIWideBoxButton/UIWideBoxButton.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { UILayoutConstant } from '@tonlabs/uikit.layout';
+import { StyleSheet, View } from 'react-native';
+import { Pressable } from '../Pressable';
+import type { UIWideBoxButtonProps } from './types';
+import { UIWideBoxButtonSecondary } from './content/UIWideBoxButtonSecondary';
+import { Caption } from './Caption';
+import { UIWideBoxButtonType } from './constants';
+import { UIWideBoxButtonPrimary } from './content/UIWideBoxButtonPrimary';
+import { UIWideBoxButtonNulled } from './content/UIWideBoxButtonNulled';
+
+export function UIWideBoxButton(props: UIWideBoxButtonProps) {
+    const {
+        type = UIWideBoxButtonType.Primary,
+        caption,
+        onPress,
+        disabled,
+        layout,
+        loading,
+        testID,
+    } = props;
+
+    const Content = React.useMemo(
+        function Content() {
+            switch (type) {
+                case UIWideBoxButtonType.Secondary:
+                    return UIWideBoxButtonSecondary;
+                case UIWideBoxButtonType.Nulled:
+                    return UIWideBoxButtonNulled;
+                case UIWideBoxButtonType.Primary:
+                default:
+                    return UIWideBoxButtonPrimary;
+            }
+        },
+        [type],
+    );
+    if (type === UIWideBoxButtonType.Nulled) {
+        return (
+            <Pressable
+                testID={testID}
+                disabled={disabled}
+                loading={loading}
+                onPress={onPress}
+                style={[style.container, layout]}
+            >
+                <Content {...props} />
+                <Caption title={caption} wideBoxButtonType={type} />
+            </Pressable>
+        );
+    }
+    return (
+        <View style={style.container}>
```

Comment:
Can you please add a comment why it's needed. As it's actually not clear for me :)

Reply:
Just not to forget and not delete in future

Reply:
Actually it doesn't necessary anymore. Removed